### PR TITLE
Fixes: #234 - customInit: Can't find config setting

### LINKED ISSUE #234: customInit: Can't find config setting
**Version of Helm and Kubernetes**:

* Kustomize Version: v4.5.4
* Kubernetes Version: v1.26.9+07c0911
* zammad-10.1.0

**What happened**:
Using the new customInit from #231

```
zammadConfig:
  initContainers:
    zammad:
      customInit: bundle exec rails r "Setting.set(\"user_email_multiple_use\", \"true\")"
```

leads to a correct ConfigMao
```

$ oc get cm zammad-init -o yaml | grep email -B2
        echo "${AUTOWIZARD_JSON}" | base64 -d > /opt/zammad/var/auto_wizard.json
    fi
    bundle exec rails r "Setting.set(\"user_email_multiple_use\", \"true\")"
```

but fails in zammad-init with `Can't find config setting`.
```
$ oc logs zammad-1 -c zammad-init -f
I, [2023-11-21T12:55:52.258560 #7]  INFO -- : ActionCable is using the redis instance at redis://:zammad@zammad-redis-master:6379.
I, [2023-11-21T12:55:52.284601#7-5520]  INFO -- : Using memcached as Rails cache store.
I, [2023-11-21T12:55:52.284888#7-5520]  INFO -- : Using the Redis back end for Zammad's web socket session store.
/opt/zammad/app/models/setting.rb:38:in `set': Can't find config setting 'user_email_multiple_use' (RuntimeError)
        from /usr/local/bundle/gems/railties-6.1.7.6/lib/rails/commands/runner/runner_command.rb:45:in `<main>'
        from /usr/local/bundle/gems/railties-6.1.7.6/lib/rails/commands/runner/runner_command.rb:45:in `eval'
        from /usr/local/bundle/gems/railties-6.1.7.6/lib/rails/commands/runner/runner_command.rb:45:in `perform'
```

**What you expected to happen**:

same as with exec
```
$ oc exec zammad-0 -c zammad-railsserver -- rails r 'p Setting.set("user_email_multiple_use", true)'
I, [2023-11-21T13:00:12.062167 #24]  INFO -- : ActionCable is using the redis instance at redis://:zammad@zammad-redis-master:6379.
I, [2023-11-21T13:00:12.114569#24-5520]  INFO -- : Using memcached as Rails cache store.
I, [2023-11-21T13:00:12.114948#24-5520]  INFO -- : Using the Redis back end for Zammad's web socket session store.
I, [2023-11-21T13:00:18.794269#24-5520]  INFO -- : Setting.set('user_email_multiple_use', true)
true
```

### PR DESCRIPTION
**Pull request by mgruner**
#### Which issue this PR fixes

- fixes #234 

#### Special notes for your reviewer

- The solution was basically to first execute `postgresql-init`, and only then `zammad-init`, which also needed the database envs to be able to run Rails commands.
- toYaml seemed to break multiline statements, so I removed it - not entirely sure

#### Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [x] Chart Version bumped

### CODE REVIEW

File: zammad/templates/configmap-init.yaml [hunk 1]
```diff
@@ -37,6 +37,6 @@ data:
         echo "${AUTOWIZARD_JSON}" | base64 -d > /opt/zammad/var/auto_wizard.json
     fi
       {{- with .Values.zammadConfig.initContainers.zammad.customInit }}
-        {{- toYaml . | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (monotek) [reply zammad/templates/configmap-init.yaml:40]: sorry, my fault. 
i just had copied it from somewhere else and did not tested it...

**Review by monotek — COMMENTED** [review]
i wonder why zammad-init is not the last one anyway?
maybe elasticsearch init should be moved above too?

**Comment by mgruner** [comment]
> i wonder why zammad-init is not the last one anyway? maybe elasticsearch init should be moved above too?

I believe the order makes sense now:

- `postgresql-init` handles database installation and migration.
- `zammad-init` handles application (pre)configuration. This might also affect the next step:
- `elasticsearch-init` rebuilds the ES index.

From my pov this should work and is also kindof similar to how `zammad-init` works in zammad-docker-compose. Of course we can add a `customInit` also to `elasticsearch-init` later if it turns out to be required. Ok?

**Comment by monotek** [comment]
Or we're making it an own init step?

**Comment by mgruner** [comment]
For me the present proposal seems to be sufficient at this time. Of course, by mounting custom configMaps, everything can be overridden for special scenarios.

**Comment by mgruner** [comment]
We cannot run it before `postgresql-init` because this handles the seeding of the database, which is required to store Settings.

**Review by monotek — APPROVED** [review]

**Pull request merged**